### ISSUE TITLE
docs: popover/anchor dropdown examples should note focus accessibility

### ISSUE THREAD

**Issue by jakeisonline** [saadeghi/daisyui#4127]
### On which page do you see this issue?

https://daisyui.com/components/dropdown/

### Describe the issue

The example for ["Dropdown using popover API and anchor positioning"](https://daisyui.com/components/dropdown/#dropdown-using-popover-api-and-anchor-positioning) has an accessibility problem. The menu items cannot be selected or navigated using only keyboard interaction.

The following example is given in the documentation:

```html
<button class="btn" popovertarget="popover-1" style="anchor-name:--anchor-1">
  Button
</button>
<ul class="dropdown menu w-52 rounded-box bg-base-100 shadow-sm"
  popover id="popover-1" style="position-anchor:--anchor-1">
  <li><a>Item 1</a></li>
  <li><a>Item 2</a></li>
</ul>
```
This example will not be keyboard navigable.

MDN's documentation on ["Popover accessibility features"](https://developer.mozilla.org/en-US/docs/Web/API/Popover_API/Using#popover_accessibility_features) states:

> When the popover is shown, the keyboard focus navigation order is updated so that the popover is next in the sequence: for example, when a button is pressed to show a popover, any buttons inside the popover will be next in the tabbing order (will be focused by pressing the Tab key).

Thankfully, the logic here considers an anchor tag _with a href defined_ as a button.

Simply, adding an `href`, even if blank, will fix the lack of keyboard focus.

```diff
<ul class="dropdown menu w-52 rounded-box bg-base-100 shadow-sm"
  popover id="popover-1" style="position-anchor:--anchor-1">
-  <li><a>Item 1</a></li>
+  <li><a href>Item 1</a></li>
-  <li><a>Item 2</a></li>
+  <li><a href>Item 2</a></li>
</ul>
```

Could this be something to note in the docs and example code? 

> [!NOTE]  
> This issue is not present on the other examples on this page, largely because the "Preview" tab code uses actual `<button>`, despite the "HTML" tab code using `<a>`, which seems perhaps problematic.

### What browsers are you seeing the problem on?

Chrome

**Comment by saadeghi** [comment]
This is not about popover.  
In HTML a `<a>` tag is not focusable without `href`. It shouldn't be! because why do you want to focus something that is not clickable 😅  
I assume in a real project you are going to use actual links with `href`.  
The code `<li><a>Item 1</a></li>` is for demo. because I don't know where do you want to link this item to. That's something you should decide, and when you add your `href` it will be usable and accessible.
